demos/python_demos/face_recognition_demo.py

```diff
@@ -0,0 +1,1037 @@
+        self.input_blob = next(iter(model.inputs))
+        self.output_blob = next(iter(model.outputs))
+        self.input_shape = model.inputs[self.input_blob].shape
+
+        assert np.array_equal([1, 3, 48, 48], self.input_shape), \
+            "Expected model input shape %s, but got %s" % \
+            ([1, 3, 48, 48], self.input_shape)
+
+        assert np.array_equal([1, self.POINTS_NUMBER * 2, 1, 1],
+                model.outputs[self.output_blob].shape), \
+            "Expected model output shape %s, but got %s" % \
+            ([1, self.POINTS_NUMBER * 2, 1, 1],
+             model.outputs[self.output_blob].shape)
+
+    def preprocess(self, frame, rois):
+        frame = self._adjust_color(frame)
+        inputs = cut_rois(frame, rois)
+        inputs = [self._resize(input, self.input_shape) for input in inputs]
+        return inputs
+
+    def enqueue(self, input):
+        # Expected input: (1, 3, 48, 48) RGB
+        assert np.array_equal(self.input_shape, input.shape), \
+            "Expected input shape %s, got %s" % (self.input_shape, input.shape)
+        return super(LandmarksDetector, self).enqueue({self.input_blob: input})
+
+    def start_async(self, frame, rois):
+        inputs = self.preprocess(frame, rois)
+        for input in inputs:
+            self.enqueue(input)
+
+    def get_landmarks(self):
+        outputs = self.get_outputs()
+        results = [ self.Result(out[self.output_blob].reshape((-1, 2))) \
+                    for out in outputs ]
+        return results
+
+    def _adjust_color(self, frame):
+        assert len(frame.shape) == 4 and frame.shape[1] == 3, \
+            "Expected input of shape [1, 3, h, w] in BGR format"
+        image = frame[0]
+        return np.array([[image[2], image[1], image[0]]]) # convert BGR to RGB
+
+
+class FacesDatabase:
+    IMAGE_EXTENSIONS = ['.jpg', '.png']
+
+    class Identity:
+        def __init__(self, label, descriptor):
+            self.label = label
+            self.descriptor = descriptor
+
+    class ROI:
+        def __init__(self, x, y, w, h):
+            self.data = np.array((x, y, w, h))
+            self.position = self.data[0:2]
+            self.size = self.data[2:4]
+
+    def __init__(self, path, face_identifier, landmarks_detector):
+        path = osp.abspath(path)
+        paths = []
+        if osp.isdir(path):
+            ext = self.IMAGE_EXTENSIONS
+            paths = [osp.join(path, f) for f in os.listdir(path) \
+                      if f.endswith(ext[0]) or f.endswith(ext[1])]
+        else:
+            raise Exception("Wrong face images database path. Expected path to" \
+                            "a directory containing '%s' files, got '%s'" % \
+                            (self.IMAGE_EXTENSIONS, path))
+
+        self.database = []
+        for path in paths:
+            label = osp.splitext(osp.basename(path))[0]
+            image = cv2.imread(path)
+
```

Comment:
It's better to assume that as an input you have **_uncropped_**  image with face, so, first you need to run face detector, second  - landmarks regressor and then face identifier in order to obtain correct embedding for the face in gallery.